### PR TITLE
Matching service

### LINKED ISSUE #16: Assignment 5 (10 points)
### Requirements
- [x] #23
- [x] #24
- [x] #25
- [x] Containerizing the application [2]

### PR DESCRIPTION
**Pull request by cjyothika**
Completed Frontend and Backend for matching-service
- User is able to find a match based on difficulty and language
- User can view the length of the queue based on the time he loads the browsers or sends a match request or receives a match

Things to work on:
- Display username instead of userid for match
- Dequeue user when browser is closed or he logs off
- UI aesthetic improvements

### CODE REVIEW

File: frontend/src/app/matching/matching.component.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  constructor(private matchingService: MatchingService, private storageService: StorageService){}
+
+  ngOnInit() {
+    this.getQueueLength();
+  }
+
+  getMatch(userDetails: any) {
+    this.matchingService.enqueue(userDetails).pipe(timeoutWith(2000, throwError(() => {
+      this.matchingService.dequeue(userDetails["userid"]).subscribe((res) => {
```
Comment (GenFusion122) [inline frontend/src/app/matching/matching.component.ts:28]: So is this subscribe stuff the rxjs, how does it work exactly? Does it just wait for a res from the server and like throw an error if it times out?

**Review by GenFusion122 — DISMISSED** [review]
LGTM! But I left a comment asking a question on the code.

**Comment by chia-yh** [comment]
LGTM, however, try and link to the specific requirements that are fulfilled, then the assignment as well if everything is closed

**Review by chia-yh — APPROVED** [review]

**Review by GenFusion122 — APPROVED** [review]
LGTM!

File: frontend/src/app/matching/matching.component.ts (hunk 1 above)
Comment (cjyothika) [reply frontend/src/app/matching/matching.component.ts:28]: So basically it's a socket I think that waits for an HTTP response from the URL. It does not timeout unless you specify it like I did here.

**Pull request merged**